esphome/components/sgp30/sensor.py

```diff
@@ -0,0 +1,55 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import i2c, sensor
+from esphome.const import CONF_ID, ICON_RADIATOR, UNIT_PARTS_PER_MILLION, \
+    UNIT_PARTS_PER_BILLION, CONF_HUMIDITY, CONF_TEMPERATURE, ICON_PERIODIC_TABLE_CO2
+
+DEPENDENCIES = ['i2c']
+
+sgp30_ns = cg.esphome_ns.namespace('sgp30')
+SGP30Component = sgp30_ns.class_('SGP30Component', cg.PollingComponent, i2c.I2CDevice)
+
+CONF_ECO2 = 'eco2'
+CONF_TVOC = 'tvoc'
+CONF_BASELINE = 'baseline'
+CONF_UPTIME = 'uptime'
+CONF_COMPENSATION = 'compensation'
+CONF_COMPENSATION_HUMIDITY = 'humidity_source'
+CONF_COMPENSATION_TEMPERATURE = 'temperature_source'
+
+CONFIG_SCHEMA = cv.Schema({
+    cv.GenerateID(): cv.declare_id(SGP30Component),
+    cv.Required(CONF_ECO2): sensor.sensor_schema(UNIT_PARTS_PER_MILLION,
+                                                 ICON_PERIODIC_TABLE_CO2, 0),
+    cv.Required(CONF_TVOC): sensor.sensor_schema(UNIT_PARTS_PER_BILLION, ICON_RADIATOR, 0),
+    cv.Optional(CONF_BASELINE): cv.hex_uint16_t,
+    cv.Optional(CONF_COMPENSATION): cv.Schema({
+        cv.Optional(CONF_COMPENSATION_HUMIDITY): cv.use_id(sensor.Sensor),
+        cv.Optional(CONF_COMPENSATION_TEMPERATURE): cv.use_id(sensor.Sensor)
```

Comment:
```suggestion
        cv.Required(CONF_COMPENSATION_TEMPERATURE): cv.use_id(sensor.Sensor)
```

Make these two fields required so that *if* the user has a `compensation` block, they need to supply both humidity and temperature sources.